src/main/java/io/kestra/plugin/cloudquery/Sync.java

```diff
@@ -0,0 +1,214 @@
+            code = {
+                "        env:",
+                "          AWS_ACCESS_KEY_ID: \"{{ secret('AWS_ACCESS_KEY_ID') }}\"",
+                "          AWS_SECRET_ACCESS_KEY: \"{{ secret('AWS_SECRET_ACCESS_KEY') }}\"",
+                "          AWS_DEFAULT_REGION: \"{{ secret('AWS_DEFAULT_REGION') }}\"    ",
+                "          PG_CONNECTION_STRING: \"postgresql://postgres:{{secret('DB_PASSWORD')}}@host.docker.internal:5432/demo?sslmode=disable\"",
+                "        configs:",
+                "           - sources.yml",
+                "           - destination.yml",
+            }
+        ),
+    }
+)
+public class Sync extends AbstractCloudQueryCommand implements RunnableTask<ScriptOutput> {
+    private static final ObjectMapper OBJECT_MAPPER = JacksonMapper.ofYaml();
+    private static final String DB_FILENAME = "icrementaldb.sqlite";
+    private static final String CLOUD_QUERY_STATE = "CloudQueryState";
+
+    @Schema(
+        title = "CloudQuery configurations.",
+        description = "A list of CloudQuery configurations or files containing CloudQuery configurations"
+    )
+    @PluginProperty(
+        dynamic = false,
+        additionalProperties = String.class
+    )
+    @NotNull
+    private Object configs;
+
+    @Schema(
+        title = "Use kestra internal backend to save incremental index.",
+        description = "Kestra can automatically add a backend option to your sources and same incremental indexes in the internal storage." +
+            "Use this boolean to activate this option."
+    )
+    @PluginProperty
+    @Builder.Default
+    private boolean incremental = false;
+
+    @Override
+    public ScriptOutput run(RunContext runContext) throws Exception {
+        CommandsWrapper commands = new CommandsWrapper(runContext)
+            .withWarningOnStdErr(true)
+            .withRunnerType(RunnerType.DOCKER)
+            .withDockerOptions(injectDefaults(getDocker()))
+            .withEnv(this.getEnv());
+        Path workingDirectory = commands.getWorkingDirectory();
+
+        File incrementalDBFile = new File(workingDirectory + "/" + DB_FILENAME);
+        try {
+            InputStream taskCacheFile = runContext.getTaskStateFile(CLOUD_QUERY_STATE, DB_FILENAME);
+            Files.copy(taskCacheFile, incrementalDBFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        } catch (FileNotFoundException exception) {
+            if (!incrementalDBFile.createNewFile()) {
+                throw new IOException("Unable to create incremental backend file.");
+            }
+        }
+
+        Map<String, Object> backendOptionsObject = getBackendOptionObject();
+        List<Map<String, Object>> configs = readConfigs(runContext, this.configs, backendOptionsObject);
+        if (incremental) {
+            configs.add(getIncrementalSqliteDestination());
+        }
+
+
+        StringBuilder command = new StringBuilder("cloudquery sync");
```

Comment:
```suggestion
        StringBuilder command = new StringBuilder("sync");
```
Use only sync command to avoid having to play with env vars